types/styled-system/index.d.ts

```diff
@@ -36,17 +36,20 @@ export interface styleFn {
     propTypes?: string[];
 }
 
-export interface LowLevelStylefunctionArguments {
-    prop: string;
-    cssProperty?: string;
-    alias?: string;
-    key?: string;
-    transformValue?: (n: string | number, scale: Array<string | number>) => any;
-    scale?: Array<string | number>;
+export type Scale = object | Array<string | number>;
```

Comment:
Do those “nested objects” not follow a particular shape?

Reply:
Scales aren't known upfront, so should be generic below too actually. (we use deeply nested objects for example).